portfolio/src/main/java/com/google/sps/servlets/EditDataServlet.java

```diff
@@ -41,38 +45,72 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Key key = KeyFactory.createKey("Comment", Long.parseLong(request.getParameter("key")));
     String type = request.getParameter("type");
+    long userID = Long.parseLong(request.getParameter("id"));
     
-    Query query = new Query("Comment", key);
+    Query keyQuery = new Query("Comment", key);
+    FilterPredicate filter = new FilterPredicate("id", Query.FilterOperator.EQUAL, userID);
+    Query idQuery = new Query("ID");
+    idQuery.setFilter(filter);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     // PreparedQuery that contains all the comments inside it
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery commResults = datastore.prepare(keyQuery);
+    // PreparedQuery that has the user who liked the comment inside it
+    PreparedQuery userResults = datastore.prepare(idQuery);
 
     // Create Logger for warning reporting
     Logger logger = Logger.getLogger(EditDataServlet.class.getName());
     logger.setLevel(Level.WARNING); 
 
-    for (Entity entity : results.asIterable()) {
-      if (type.equals("like")) {
-        Object likes = entity.getProperty("likes");
-        if (likes instanceof Long) {
-          entity.setProperty("likes", ((long) likes) + 1);
-        }
-        else {
-          logger.warning("Could not convert Entity's likes to long"); 
-        }
+    for (Entity entity : commResults.asIterable()) {
```

Comment:
Maybe name the loop variable `comment` instead of `entity`

Reply:
Got it!